interp/impargs.ml

```diff
@@ -499,9 +499,9 @@ type implicit_interactive_request =
 
 type implicit_discharge_request =
   | ImplLocal
-  | ImplConstant of Constant.t * implicits_flags
+  | ImplConstant of implicits_flags
   | ImplMutualInductive of MutInd.t * implicits_flags
-  | ImplInteractive of GlobRef.t * implicits_flags *
+  | ImplInteractive of implicits_flags *
```

Comment:
Why can't the mutind be removed even though the other cases can remove the globref?

Reply:
In the other two, the reference is contained in the paired `(GlobRef.t * implicits_list list) list` as the first component of the only element of the list, while for `ImplMutualInductive`, the references in that list are to type families and constructors.
In `rebuild_implicits`, we need the reference to the whole mutual inductive block to recalculate default implicits for the whole block at once, rather than calling recalling `compute_mib_implicits` for each constructor.

(As a side note, the only shared computation in `compute_mib_implicits` is the computation of `env_ar`. It might be worthwhile in a future PR to benchmark splitting this function into two that compute implicits for one inductive type and one constructor, as this would allow completely unifying `ImplConstant` and `ImplMutualInductive`, and might not be too expensive)

Reply:
>the references in that list are to type families and constructors

It's quite easy to get the mutind from any of those though.

Reply:
Yes, it is. However, doing so felt like it would make the code more complicated, rather than simpler.